codec-base/src/main/java/io/netty/handler/codec/DefaultHeaders.java

```diff
@@ -988,8 +988,8 @@ public final int hashCode(HashingStrategy<V> valueHashingStrategy) {
         for (K name : names()) {
             result = 31 * result + hashingStrategy.hashCode(name);
             List<V> values = getAll(name);
-            for (int i = 0; i < values.size(); ++i) {
-                result = 31 * result + valueHashingStrategy.hashCode(values.get(i));
+            for (V value : values) {
```

Comment:
iterator allocation introduced. you need to keep get(i), please revert this and similar changes. Also this PR is very big with many different refactorings. I recommend you to split it into multiple smaller one, 1 pr - 1 kind of refactorings so it's easier to review.

Reply:
@doom369 I've reverted all changes with for loops. No iterator allocations anymore.

P.S. Generally I agree that PR could be split to 2-3 smaller commits. 

BUT
The PR may seem big in terms of LOC. But it's not big in terms of "different refactorings". These all are very simple, automated refactorings. Mostly simplification of "equals" and "hashCode" methods.